src/components/Datatable/Datatable.tsx

```diff
@@ -134,17 +140,28 @@ const Datatable = <D extends Record<string, unknown>>({
     [onDataFetch, appliedFilters, searchQuery],
   );
 
+  const handleOnSuggestionsFetch = useCallback(
+    async (query: string) => {
+      setFilterSuggestions(await onSuggestionsFetch(query));
+    },
+    [onSuggestionsFetch],
+  );
+
   const handleOnSearch = useCallback(
-    (query: string) => {
-      setSearchQuery(query);
-      onDataFetch({
-        pageSize: defaultPageSize,
-        pageIndex: 0,
-        query,
-        filters: appliedFilters,
-      });
+    async (queryValue: string) => {
+      setSearchQuery(queryValue);
+      if (controlsConfig.searchConfig?.hasSuggestions === true) {
+        handleOnSuggestionsFetch(queryValue);
+      } else {
+        onDataFetch({
+          pageSize: defaultPageSize,
+          pageIndex: 0,
+          query: searchQuery,
+          filters: appliedFilters,
+        });
+      }
     },
-    [defaultPageSize, onDataFetch, appliedFilters],
+    [onDataFetch, defaultPageSize, searchQuery, appliedFilters, controlsConfig],
```

Comment:
remove `searchQuery` and do not use objects (`controlsConfig`) in dependency array, rather destructure and use simple types, cause any change of object will cause re-initialization of this function

Reply:
also you are missing `handleOnSuggestionsFetch` in dependency array